app/questionnaire/router.py

```diff
@@ -179,57 +200,64 @@ def is_survey_complete(self):
 
         return True
 
-    def is_path_complete(self, path):
-        location = self._get_first_incomplete_location(path)
+    def is_path_complete(self, routing_path_block_ids):
+        location = self._get_first_incomplete_location(routing_path_block_ids)
         if not location or (
-            location == path[-1]
+            location.block_id == routing_path_block_ids[-1]
             and self._schema.get_block(location.block_id)["type"] == "SectionSummary"
         ):
             return True
-
         return False
 
     def get_section_return_location_when_section_complete(
         self, routing_path
     ) -> Location:
-        return self._get_location_of_section_summary(routing_path) or routing_path[0]
+        return self._get_location_of_section_summary(routing_path) or Location(
+            block_id=routing_path[0],
+            section_id=routing_path.section_id,
+            list_item_id=routing_path.list_item_id,
+            list_name=routing_path.list_name,
+        )
 
     def full_routing_path(self):
-        path = []
+        full_routing_path = []
         for section_id in self.enabled_section_ids:
             repeating_list = self._schema.get_repeating_list_for_section(section_id)
 
             if repeating_list:
                 for list_item_id in self._list_store[repeating_list].items:
-                    path = path + list(
+                    full_routing_path.append(
                         self._path_finder.routing_path(
                             section_id=section_id, list_item_id=list_item_id
                         )
                     )
             else:
-                path = path + list(
+                full_routing_path.append(
                     self._path_finder.routing_path(section_id=section_id)
                 )
+        return full_routing_path
 
-        return path
-
-    def _is_location_complete(self, location):
+    def _is_location_complete(self, block_id, section_id, list_item_id):
         completed_block_ids = self._progress_store.get_completed_block_ids(
-            location.section_id, location.list_item_id
+            section_id, list_item_id
         )
 
-        return location.block_id in completed_block_ids
+        return block_id in completed_block_ids
```

Comment:
This method now has nothing to do with `Location` so I think we should name it something like `_is_block_complete`.